tests/wasmfs/wasmfs_proxiedfile.c

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2021 The Emscripten Authors.  All rights reserved.
+ * Emscripten is available under two separate licenses, the MIT license and the
+ * University of Illinois/NCSA Open Source License.  Both these licenses can be
+ * found in the LICENSE file.
+ */
+
+#include <assert.h>
+#include <emscripten/emscripten.h>
+#include <emscripten/wasmfs.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <signal.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+void write_and_read(const char* msg, int fd) {
+  char buf[200] = {};
+  errno = 0;
+  write(fd, msg, strlen(msg));
+  assert(errno == 0);
+  errno = 0;
+  lseek(fd, 0, SEEK_SET);
+  assert(errno == 0);
+  read(fd, buf, sizeof(buf));
+  assert(errno == 0);
+  printf("%s\n", buf);
+}
+
+int main() {
```

Comment:
This test is very similar to the one for the JS backend. Can we combine them into a single parameterized test using `-D` flags on the compilation command?